pyfr/solvers/navstokes/kernels/avis.mako

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+<%inherit file='base'/>
+<%namespace module='pyfr.backends.base.makoutil' name='pyfr'/>
+
+<%pyfr:kernel name='avis' ndim='1'
+              s='out view fpdtype_t[${str(nupts)}]'
+              amu_e='out view fpdtype_t[${str(nupts)}]'
+              amu_f='out view fpdtype_t[${str(nfpts)}]'>
+    // Smoothness indicator
+    fpdtype_t totEn = 0.0, pnEn = 0.0, s2;
+
+% for i in range(nupts):
+    s2 = s[${i}]*s[${i}];
+    totEn += s2;
+
+% if ubdegs[i] >= order:
+    pnEn += s2;
+% endif
+% endfor
+
+    // Prevent log(0) for uniform flow
+    fpdtype_t se = (pnEn > 0.0) ? log10(pnEn/totEn) : log10(${c['eps']});
```

Comment:
I tried to 'eps' to base.mako using #define (backend level), but I cannot use it in python (pyfr-mako). I will make the function to compute machine epsilon in pyfr.nputils, which can be called in pyfr.backend.base.makoutil. I will commit this, as well as other fixes, shortly. Please let me know if you think a better way to do it.